crates/wasi-common/src/hostcalls/sock.rs

```diff
@@ -41,3 +43,25 @@ pub unsafe fn sock_shutdown(
 ) -> wasi::__wasi_errno_t {
     unimplemented!("sock_shutdown")
 }
+
+hostcalls! {
+    pub unsafe fn sock_connect(
+        wasi_ctx: &WasiCtx,
+        memory: &mut [u8],
+        sock: wasi::__wasi_fd_t,
+        addr_ptr: wasi32::uintptr_t,
+        addr_len: wasi32::size_t,
+    ) -> wasi::__wasi_errno_t;
+
+    pub unsafe fn sock_socket(
+        wasi_ctx: &mut WasiCtx,
+        memory: &mut [u8],
+        sock_domain: i32,
+        // socket type
+        // DGRAM 5
+        // STREAM 6
+        sock_type: wasi::__wasi_filetype_t,
+        sock_protocol: i32,
+        fd_out_ptr: wasi32::uintptr_t,
```

Comment:
I think you're missing `rights` field here based on the prototype of this syscall in `wasi-misc-tests`.